src/com/intellij/idea/plugin/hybris/runConfigurations/LocalSapCXRunSettingsEditor.kt

```diff
@@ -0,0 +1,147 @@
+import com.intellij.openapi.observable.util.bind
+import com.intellij.ui.components.fields.ExtendableTextField
+import com.intellij.ui.components.fields.IntegerField
+import org.jetbrains.idea.maven.execution.MavenRunConfiguration
+import org.jetbrains.idea.maven.project.MavenConfigurableBundle
+
+class LocalSapCXRunSettingsEditor(val runConfiguration: LocalSapCXRunConfiguration) :
+    RunConfigurationFragmentedEditor<LocalSapCXRunConfiguration>(runConfiguration, JavaRunConfigurationExtensionManager()) {
+
+    override fun createRunFragments(): List<SettingsEditorFragment<LocalSapCXRunConfiguration, *>> = SettingsEditorFragmentContainer.fragments {
+        add(CommonParameterFragments.createRunHeader())
+        addBeforeRunFragment(CompileStepBeforeRun.ID)
+        addAll(BeforeRunFragment.createGroup())
+        addDebugHostFragment()
+        addDebugPortFragment()
+        addDebugLineFragment(runConfiguration.getRemoteConnetion().launchCommandLine)
+        add(LogsGroupFragment())
+        addEnvironmentFragment()
+    }
+    private fun SettingsEditorFragmentContainer<LocalSapCXRunConfiguration>.addEnvironmentFragment() =
+        addEnvironmentFragment(
+            object : LabeledSettingsFragmentInfo {
+                override val editorLabel: String = ExecutionBundle.message("environment.variables.component.title")
+                override val settingsId: String = "maven.environment.variables.fragment"
+                override val settingsName: String = ExecutionBundle.message("environment.variables.fragment.name")
+                override val settingsGroup: String = MavenConfigurableBundle.message("maven.run.configuration.runner.options.group")
+                override val settingsHint: String = ExecutionBundle.message("environment.variables.fragment.hint")
+                override val settingsActionHint: String = ExecutionBundle.message("set.custom.environment.variables.for.the.process")
+            },
+            { sapCXOptions.environmentProperties },
+            { sapCXOptions.environmentProperties = it as MutableMap<String, String> },
+            { sapCXOptions.isPassParentEnv },
+            { sapCXOptions.isPassParentEnv = it },
+            hideWhenEmpty = true
+        )
+    private fun SettingsEditorFragmentContainer<LocalSapCXRunConfiguration>.addDebugHostFragment() = addLabeledSettingsEditorFragment(
+        object : LabeledSettingsFragmentInfo {
+            override val editorLabel: String = HybrisI18NBundleUtils.message("hybris.project.run.configuration.localserver.host.field")
+
+            override val settingsId: String = "hybris.localserver.debug.host.fragment"
+            override val settingsName: String = "remoteDebugHost"
+            override val settingsGroup: String? = null
+            override val settingsHint: String = ""
+            override val settingsActionHint: String? = null
+        },
+        { HostField(HybrisConstants.REMOTE_DEBUG_DEFAULT_HOST) },
```

Comment:
it looks like hardcode. please consider more advanced configuration / identification in the next PR

Reply:
This is only to provide Default Values and placeholders. The Value is still changable. See Image in the First PR Comment